lib/manageiq/release/pull_request_blaster_outer.rb

```diff
@@ -0,0 +1,136 @@
+require 'pathname'
+
+module ManageIQ
+  module Release
+    class PullRequestBlasterOuter
+      attr_reader :repo, :branch, :script, :dry_run, :message
+
+      ROOT_DIR = Pathname.new(__dir__).join("..", "..", "..").freeze
+      def initialize(repo, branch:, script:, dry_run:, message:)
+        @repo    = repo
+        @branch  = branch
+        @script  = begin
+          s = Pathname.new(script)
+          s = ROOT_DIR.join(script) if s.relative?
+          raise "File not found #{s}" unless File.exist?(s)
+          s.to_s
+        end
+        @dry_run = dry_run
+        @message = message
+      end
+
+      def blast
+        puts "+++ blasting #{repo.github_repo}..."
+
+        repo.git
+        repo.fetch(output: false)
+        repo.checkout(pr_branch, "origin/#{branch}")
+
+        run_script
+
+        result = false
+        if !commit_changes
+          puts "!!! Failed to commit changes. Perhaps the script is wrong or #{repo.github_repo} is already updated."
+        elsif dry_run
+          result = "Committed but is dry run"
+        else
+          puts "Do you want to open a pull request on #{repo.github_repo} with the above changes? (Y/N)"
+          answer = $stdin.gets.chomp
+          if answer.upcase.start_with?("Y")
+            fork_repo unless forked?
+            push_branch
+            result = open_pull_request
+          end
+        end
+        puts "--- blasting #{repo.github_repo} complete"
+        result
+      end
+
+      private
+
+      def github
+        ManageIQ::Release.github
+      end
+
+      def forked?
+        github.repos(github.login).any? { |m| m.name == repo.name }
+      end
+
+      def with_status
+        calling_method = caller_locations(1,1)[0].label
+        puts "+++ #{calling_method}..."
+        result = yield
+        puts "--- #{calling_method}: #{result}"
+        result
+      end
+
+      def fork_repo
+        with_status do
+          github.fork(repo.github_repo)
+          until forked?
+            print "."
+            sleep 3
+          end
+        end
+      end
+
+      def run_script
+        with_status do
+          Dir.chdir(repo.path) do
+            `#{script}`
+          end
+        end
+      end
+
+      def commit_changes
+        with_status do
+          Dir.chdir(repo.path) do
+            begin
+              repo.git.add("-v", ".")
+              repo.git.commit("-m", message)
+              repo.git.show
+              if dry_run
+                puts "!!! --dry-run enabled: If the above commit in #{repo.path} looks good, run again without dry run to fork the repo, push the branch and open a pull request."
+              end
+              true
+            rescue MiniGit::GitError => e
+              e.status.exitstatus == 0
+            end
+          end
+        end
+      end
+
+      def origin_remote
```

Comment:
For @bdunne, I named this "origin_remote" for us normal people knew

Reply:
typing is hard

Reply:
@Fryguy Naming is hard.  Suggestions welcome.  `remote4fork` ??? :trollface:

Reply:
@jrafanie I honestly have no idea what this is means :laughing: